### DISCUSSION TITLE
Lack of javascript error messages with webpack when missing a dependency

### DISCUSSION THREAD

**Discussion by arcanefoam** [webpack/webpack#16251] [Q&A]
## Feature request

**What is the expected behavior?**

To get an error message when a dependency is not found at runtime via the webpackBootstrap.

**What is motivation or use case for adding/changing the behavior?**

Currently this functionality is not available. The problem is that compiling and bundling works. But once you load your page, the javascript does not execute and it just silently fails. This make its really HARD to determine what is the cause. I have been running around in circles for three days with this, thinking it was an issue with my ts code, ts configuration or webpack confguration.

**How should this be implemented in your opinion?**
I am using the 'extract' option via Laravel MIx, so I get a manifest.js that has the  webpackBootstrap code. I hava emodified the code so it shows an error when a chunk can't be found. This is the relevant section:

```
/* webpack/runtime/chunk loaded */
/******/ 	(() => {
/******/ 		var deferred = [];
/******/ 		__webpack_require__.O = (result, chunkIds, fn, priority) => {
/******/ 			if(chunkIds) {
/******/ 				priority = priority || 0;
/******/ 				for(var i = deferred.length; i > 0 && deferred[i - 1][2] > priority; i--) deferred[i] = deferred[i - 1];
/******/ 				deferred[i] = [chunkIds, fn, priority];
/******/ 				return;
/******/ 			}
/******/ 			var notFulfilled = Infinity;
/******/ 			for (var i = 0; i < deferred.length; i++) {
/******/ 				var [chunkIds, fn, priority] = deferred[i];
/******/ 				var fulfilled = true;
/******/ 				for (var j = 0; j < chunkIds.length; j++) {
                                                  // This breaks the  Object.keys(__webpack_require__.O).every((key) => (__webpack_require__.O[key](chunkIds[j])))
                                                 // Statement into a loop, so when a chunk is misisng we can issue the error.
							var everyChunk = true;
							const keys = Object.keys(__webpack_require__.O);
							keys.forEach((key, index) => {
								if (!__webpack_require__.O[key](chunkIds[j])) {
									console.error(`Unable to find required file: ${chunkIds[j]}. Make sure the relevant file has been added via <script> or <style> element.`)
									everyChunk = false;
								}
							});
/******/ 					if ((priority & 1 === 0 || notFulfilled >= priority) && everyChunk) {
/******/ 						chunkIds.splice(j--, 1);
/******/ 					} else {
/******/ 						fulfilled = false;
/******/ 						if(priority < notFulfilled) notFulfilled = priority;
/******/ 					}
/******/ 				}
/******/ 				if(fulfilled) {
/******/ 					deferred.splice(i--, 1)
/******/ 					var r = fn();
/******/ 					if (r !== undefined) result = r;
/******/ 				}
/******/ 			}
/******/ 			return result;
/******/ 		};
```

**Are you willing to work on this yourself?**

yes. But I am not 100% sure I completely understand the webpackBootstrap code to assert if the modification is correct.

**Comment by alexander-akait** [comment]
Please provide reproducible test repo, sounds like you forgot to include a runtime chunk

**Reply by arcanefoam** [reply]
Yes, that is exactly the problem! If I forgot to include a runtime chunk, then there is no alert, js code doesn't run and there is no indication of why. BTW this comes from a question in Stackoverflow: https://stackoverflow.com/q/67649379/3837873. The OP has provided an example repo.


I am using webpack via Laravel Mix, so I do not use the HTMLPlugin to insert all the required scripts/chunks. In these scenarios, where one is adding scripts manually, having these error reports would be life saving. Perhaps a configuration flag  to enable this error messages.

**Reply by alexander-akait** [reply]
We don't add it because it will increase a size of chunks, so big size - bad for client.

**Reply by arcanefoam** [reply]
It's just a couple of lines that go in the webpack bootstrap, so it's only added once, not per chunk.¡ (as far as I can tell).

**Reply by alexander-akait** [reply]
It is not couple lines, for big projects it will be big list...

**Reply by arcanefoam** [reply]
I think I am missing something, my change us only about 10 lines that go in the webpack bootstrap code. Are you telling me that the ´__webpack_require__.O´ code is added for each chunk? As in __webpack_require__.0, __webpack_require__.1. __webpack_require__.2.. etc? 
I am just trying to understand the issue a bit better.

**Reply by alexander-akait** [reply]
Yes, there are situations when you have runtime inside each chunks, but it makes sense to have it for `mode: "development"`, but we don't add this because most of builds have another setup for production builds...

**Reply by arcanefoam** [reply]
OK, thanks. 

I do still think just silently failing is not good. It took me days to figure out it was a webpack problem.....

**Reply by alexander-akait** [reply]
@arcanefoam We can improve docs (so other developer can google it) and pdescribe this situation, what do you think?
